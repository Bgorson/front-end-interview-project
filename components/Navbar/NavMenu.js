import {useState} from 'react';
import styles from './Navbar.module.css';
import HamburgerMenu from 'react-hamburger-menu';
export default function NavMenu() {
    const [open, setOpen]= useState(false);
    const handleClick=()=> {
        setOpen(!open);
    };
    return (
        
        <>
            <HamburgerMenu
                className={styles.hamburger}
                isOpen={open}
                menuClicked={handleClick.bind(this)}
                width={18}
                height={15}
                strokeWidth={1}
                rotate={0}
                color='white'
                borderRadius={0}
                animationDuration={0.5}
            />
            {open?
                <ul className={styles.menu}>
                    <li className={styles.item}><a className={styles.link} href="#">Stores</a></li>
                    <li className={styles.item}><a className={styles.link} href="#">Contact Us</a></li>
                </ul>:null}
            <ul className={styles.deskTopMenu}>
                <li className={styles.item}><a className={styles.link} href="#">Stores</a></li>
                <li className={styles.item}><a className={styles.link} href="#">Contact Us</a></li>
            </ul>
        </>
    );}