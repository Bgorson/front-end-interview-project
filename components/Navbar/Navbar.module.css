
.hamburger{
    margin-top:1rem;
    margin-left:1rem;
}
@media (min-width: 600px) {
    .hamburger{
        display: none;
    }
  }
  