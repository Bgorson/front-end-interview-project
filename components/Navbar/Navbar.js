import NavMenu from './NavMenu';
import styles from './Navbar.module.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faShoppingBag } from '@fortawesome/free-solid-svg-icons';

export default function Navbar() {
    return (
        <nav className={styles.navBar}>
            <NavMenu/>
            <li className={styles.logo}
            ><a className={styles.link} href="#">Foxtrot</a></li>
            <li className={styles.item}>
                <a className={styles.link} href="#">
                    <FontAwesomeIcon color= {'white'} icon={faShoppingBag}></FontAwesomeIcon></a>
            </li>
        </nav>
        
    );
}
