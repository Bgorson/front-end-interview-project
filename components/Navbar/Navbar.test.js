import renderer from 'react-test-renderer';
import Navbar from './Navbar';

it('should render correctly', () => {
    const component = renderer.create(<Navbar/>).toJSON();

    expect(component).toMatchSnapshot();
});