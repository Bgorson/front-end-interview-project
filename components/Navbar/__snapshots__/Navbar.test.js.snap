// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<nav
  className="navBar"
>
  <div
    className="hamburger"
    onClick={[Function]}
    style={
      Object {
        "height": "15px",
        "position": "relative",
        "transform": "rotate(0deg)",
        "width": "18px",
      }
    }
  >
    <span
      style={
        Object {
          "background": "white",
          "borderRadius": "0px",
          "display": "block",
          "height": "1px",
          "marginTop": "-0.5px",
          "position": "absolute",
          "transform": "translate3d(0,0,0) rotate(0)",
          "transformOrigin": "center",
          "transitionDuration": "0.5s",
          "transitionTimingFunction": "ease",
          "width": "100%",
        }
      }
    />
    <span
      style={
        Object {
          "background": "white",
          "borderRadius": "0px",
          "display": "block",
          "height": "1px",
          "marginTop": "-0.5px",
          "opacity": "1",
          "position": "absolute",
          "top": "7.5px",
          "transformOrigin": "center",
          "transitionDuration": "0.125s",
          "transitionTimingFunction": "ease-out",
          "width": "100%",
        }
      }
    />
    <span
      style={
        Object {
          "background": "white",
          "borderRadius": "0px",
          "display": "block",
          "height": "1px",
          "marginTop": "-0.5px",
          "position": "absolute",
          "transform": "translate3d(0,15px,0) rotate(0)",
          "transformOrigin": "center",
          "transitionDuration": "0.5s",
          "transitionTimingFunction": "ease",
          "width": "100%",
        }
      }
    />
  </div>
  <ul
    className="deskTopMenu"
  >
    <li
      className="item"
    >
      <a
        className="link"
        href="#"
      >
        Stores
      </a>
    </li>
    <li
      className="item"
    >
      <a
        className="link"
        href="#"
      >
        Contact Us
      </a>
    </li>
  </ul>
  <li
    className="logo"
  >
    <a
      className="link"
      href="#"
    >
      Foxtrot
    </a>
  </li>
  <li
    className="item"
  >
    <a
      className="link"
      href="#"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-shopping-bag fa-w-14 "
        color="white"
        data-icon="shopping-bag"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M352 160v-32C352 57.42 294.579 0 224 0 153.42 0 96 57.42 96 128v32H0v272c0 44.183 35.817 80 80 80h288c44.183 0 80-35.817 80-80V160h-96zm-192-32c0-35.29 28.71-64 64-64s64 28.71 64 64v32H160v-32zm160 120c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zm-192 0c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
  </li>
</nav>
`;
